scipy/spatial/transform/_rigidtransformation.pyx

```diff
@@ -0,0 +1,1899 @@
+
+        Examples
+        --------
+        >>> from scipy.spatial.transform import RigidTransformation as T
+        >>> from scipy.spatial.transform import Rotation as R
+        >>> import numpy as np
+
+        Creating from a single rotation and translation:
+
+        >>> r = R.from_euler("ZYX", [90, 30, 0], degrees=True)
+        >>> r.as_matrix()
+        array([[ 0.       , -1.,  0.        ],
+               [ 0.8660254,  0.,  0.5       ],
+               [-0.5      ,  0.,  0.8660254 ]])
+        >>> d = np.array([2, 3, 4])
+        >>> t = T.from_rottrans(r, d)
+        >>> t.rotation.as_matrix()
+        array([[ 0.       , -1.,  0.        ],
+               [ 0.8660254,  0.,  0.5       ],
+               [-0.5      ,  0.,  0.8660254 ]])
+        >>> t.translation
+        array([2., 3., 4.])
+        >>> t.single
+        True
+
+        When applying a transformation to a vector, the result is the same as
+        if the transformation was applied to the vector in the following way:
+        ``t.apply(vector) == translation + rotation.apply(vector)``
+
+        >>> r.apply([1, 0, 0])
+        array([0.       , 0.8660254, -0.5     ])
+        >>> d + r.apply([1, 0, 0])
+        array([2.       , 3.8660254,  3.5     ])
+        >>> t.apply([1, 0, 0])
+        array([2.       , 3.8660254,  3.5     ])
+        """
+        return cls.from_translation(translation) * cls.from_rotation(rotation)
+
+    @cython.embedsignature(True)
+    @classmethod
+    def from_expcoords(cls, expcoords):
+        r"""Initialize from exponential coordinates of transformation.
+
+        This implements the exponential map that converts 6-dimensional real
+        vectors to SE(3). The first three components of the vector encode
+        the rotation and the last three components encode the translation.
+
+        The exponential coordinates of transformation can be split up into
+        a screw axis :math:`\mathcal{S} \in \mathbb{R}^6` and a scalar
+        :math:`\theta` that defines the magnitude of the displacement. For
+        pure translations, the norm of the last three components defines
+        :math:`\theta` and for all other transformations, the norm of the first
+        three components defines :math:`\theta`.
+
+        Parameters
+        ----------
+        expcoords : array_like, shape (N, 6) or (6,)
+            A single exponential coordinate vector or a stack of exponential
+            coordinate vectors. The first three components define the
+            rotation and the last three components define the translation.
+
+        Returns
+        -------
+        transformation : `RigidTransformation` instance
+            A single transformation or a stack of transformations.
+
+        Examples
+        --------
+        >>> from scipy.spatial.transform import RigidTransformation as T
+        >>> import numpy as np
+
+        Creating from a single 6d vector of exponential coordinates:
```

Comment:
Might be tedious, but perhaps add more example which can give some intuition about this parametrization?

At least it should work as a rotation vector and as a simple translation when we have only rotation and only translation respectively.